cmd/namespace-lock.go

```diff
@@ -42,15 +45,21 @@ func initDsyncNodes() error {
 	// Initialize rpc lock client information only if this instance is a distributed setup.
 	clnts := make([]dsync.NetLocker, len(globalEndpoints))
 	myNode := -1
+	localLockServerIndex := 0
 	for index, endpoint := range globalEndpoints {
-		clnts[index] = newLockRPCClient(authConfig{
-			accessKey:       cred.AccessKey,
-			secretKey:       cred.SecretKey,
-			serverAddr:      endpoint.Host,
-			secureConn:      globalIsSSL,
-			serviceEndpoint: pathutil.Join(minioReservedBucketPath, lockServicePath, endpoint.Path),
-			serviceName:     lockServiceName,
-		})
+		if !endpoint.IsLocal {
```

Comment:
This also helps moving the check `if endpoint.IsLocal && mynode == -1 { myNode = index }` together to the same `if` condition.